src/components/common/YourCompany/StartDate.vue

```diff
@@ -0,0 +1,232 @@
+import { ActionBindingIF } from '@/interfaces'
+
+@Component({
+  components: {
+    DatePickerShared
+  }
+})
+export default class StartDate extends Mixins(CommonMixin, DateMixin) {
+  // Global getters
+  @Getter isBenIaCorrectionFiling!: boolean
+  @Getter isFirmCorrectionFiling!: boolean
+  @Getter getCorrectedStartDate!: string
+  @Getter getBusinessFoundingDate!: string // actually date-time
+  @Getter getCurrentDate!: string
+  @Getter getCurrentJsDate!: Date
+
+  // Global setter
+  @Action setCorrectionStartDate!: ActionBindingIF
+  @Action setValidComponent!: ActionBindingIF
+
+  /** Whether to show invalid section styling. */
+  @Prop({ default: false })
+  readonly invalidSection: boolean
+
+  protected dropdown = false
+  protected onEditMode = false
+  protected isCorrected = !!this.getCorrectedStartDate
+  protected newCorrectedStartDate = null as string // date is "Month Day, Year"
+
+  /** The minimum start date that can be entered (Up to 2 years ago today). */
+  protected get startDateMin (): Date {
+    const startDateMin = new Date(this.getCurrentJsDate)
+    startDateMin.setFullYear(startDateMin.getUTCFullYear() - 2)
+    startDateMin.setHours(0, 0, 0) // Set time to 0 for accurate Date Rules comparison
+
+    return startDateMin
+  }
+
+  /** The minimum start date string. */
+  protected get startDateMinStr (): string {
+    return this.dateToYyyyMmDd(this.startDateMin)
+  }
+
+  /** The maximum start date that can be entered (Up to 90 days from today). */
+  protected get startDateMax (): Date {
+    const startDateMax = new Date(this.getCurrentJsDate)
+    startDateMax.setDate(startDateMax.getUTCDate() + 90)
+    return startDateMax
+  }
+
+  /** The maximum start date string. */
+  protected get startDateMaxStr (): string {
+    return this.dateToYyyyMmDd(this.startDateMax)
+  }
+
+  /** The business date or business start date string. */
+  protected get businessStartDate (): string {
+    if (
+      this.isFirmCorrectionFiling ||
+      this.isFirmChangeFiling ||
+      this.isFirmConversionFiling
+    ) {
+      if (this.getCorrectedStartDate && this.isCorrected) {
+        return this.yyyyMmDdToPacificDate(this.getCorrectedStartDate)
+      } else if (this.getBusinessFoundingDate) {
+        // getBusinessFoundingDate is is stored in utc time for BEN's and COOP's
+        // For firms only the date is valid and it is in pacific time.
+        return this.yyyyMmDdToPacificDate(this.getBusinessFoundingDate.slice(0, 10))
+      }
+    }
+    return 'Unknown'
+  }
+
+  protected onChangeClicked (): void {
+    this.onEditMode = true
+  }
+
+  /** Called to add a new date. */
+  protected onOkClicked (date: string): void {
+    if (date) {
+      this.newCorrectedStartDate = date
+    }
+  }
+
+  protected onDoneClicked (): void {
+    if (this.newCorrectedStartDate) {
+      // For firms only the date is valid
+      if (
+        this.newCorrectedStartDate !==
+        this.getBusinessFoundingDate.slice(0, 10)
```

Comment:
Does the entire if statement fit on one line now?

Reply:
Yes it does.